ecoscope/io/earthranger_utils.py

```
@@ -80,5 +81,22 @@ def pack_columns(dataframe: pd.DataFrame, columns: typing.List):
     return dataframe
 
 
-def filter_bad_geojson(dataframe: pd.DataFrame):
-    return dataframe[dataframe["geojson"].apply(lambda x: True if isinstance(x, dict) and x.get("geometry") else False)]
+def geometry_from_event_geojson(
+    df: pd.DataFrame, geojson_column="geojson", force_point_geometry=True, drop_null_geometry=True
+):
+    if df.empty:
+        return gpd.GeoDataFrame()
+
+    def shape_from_geojson(geojson):
+        try:
+            result = shape(geojson.get("geometry"))
+        except Exception:
+            return None
```

Is there ever a case where the geometry is malformed and we actually want to fail hard here, so the user _knows_ it's malformed? Rather than just silently "nullifying" it?

---

So this is how it is today: currently if the `geometry` is ever malformed/such that shapely couldn't parse it, we would fail [hard](https://github.com/wildlife-dynamics/ecoscope/blob/76e25c244e17a184d4f2c1535b940773265e3d9e/ecoscope/io/earthranger.py#L638) because the conversion isn't 'safe' against that Shapely exception.

We haven't seen this scenario yet across the beta cohort, which would suggest that it probably doesn't happen very often (probably due to _how_ events in ER are collected)..

That said, we _are_ nulling out today if the geojson is ever malformed such that we can't resolve a 'geometry' key, and to me this feels like a safe extension of that. So to me it's kind of weird that we're silently nulling if the geojson is bad, but failing hard only if the geometry portion of that same geojson is bad.
I think in general since this is all happening per _event_ it's safer not to fail hard if a single event has bad geojson...